database/user.go

```diff
@@ -159,6 +159,20 @@ type (
 	}
 )
 
+// UserByAPIKey returns the user who owns the given API key.
+func (db *DB) UserByAPIKey(ctx context.Context, ak string) (*User, error) {
+	sr := db.staticAPIKeys.FindOne(ctx, bson.M{"key": ak})
```

Comment:
f/u: I wonder if there is a way to try and look up a user by both api key as well as token in a single query to reduce the number of trips to the db.

Reply:
If we find an API key in the request (no DB call so far) we try to fetch the user based on it and if we fail we return an error. So, there is no scenario in which we make two trips to the DB in order to find the authenticated user.

We *do* make two trips in case the user supplies an API key and we get them from the DB, create a token for them, and then embed the token in the context. Later on we fetch the user from the DB based on that. But this inefficiency has already been pointed out and we have a solution for it - fetch the user from the DB in `withAuth` (either by API key or by token) and embed the user object in the context instead of the token.

Based on that, I don't think we need an extra follow up here.